integration-tests/testconfig/common/vrf/common.go

```diff
@@ -0,0 +1,297 @@
+	if c.Common != nil {
+		if err := c.Common.Validate(); err != nil {
+			return err
+		}
+	}
+	if c.General != nil {
+		if err := c.General.Validate(); err != nil {
+			return err
+		}
+	}
+	if c.Performance != nil {
+		if err := c.Performance.Validate(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+type Common struct {
+	CancelSubsAfterTestRun *bool `toml:"cancel_subs_after_test_run"`
+}
+
+func (c *Common) Validate() error {
+	return nil
+}
+
+type PerformanceConfig struct {
+	TestDuration          *blockchain.StrDuration `toml:"test_duration"`
+	RPS                   *int64                  `toml:"rps"`
+	RateLimitUnitDuration *blockchain.StrDuration `toml:"rate_limit_unit_duration"`
+}
+
+func (c *PerformanceConfig) Validate() error {
+	if c.TestDuration == nil || c.TestDuration.Duration == 0 {
+		return errors.New("test_duration must be set to a positive value")
+	}
+	if c.RPS == nil || *c.RPS == 0 {
+		return errors.New("rps must be set to a positive value")
+	}
+	if c.RateLimitUnitDuration == nil {
+		return errors.New("rate_limit_unit_duration must be set ")
+	}
+
+	return nil
+}
+
+type ExistingEnvConfig struct {
+	CoordinatorAddress            *string  `toml:"coordinator_address"`
+	ConsumerAddress               *string  `toml:"consumer_address"`
+	LinkAddress                   *string  `toml:"link_address"`
+	KeyHash                       *string  `toml:"key_hash"`
+	CreateFundSubsAndAddConsumers *bool    `toml:"create_fund_subs_and_add_consumers"`
+	NodeSendingKeys               []string `toml:"node_sending_keys"`
+	Funding
+}
+
+func (c *ExistingEnvConfig) Validate() error {
+	if c.CreateFundSubsAndAddConsumers == nil {
+		return errors.New("create_fund_subs_and_add_consumers must be set ")
+	}
+	if c.CoordinatorAddress == nil {
+		return errors.New("coordinator_address must be set when using existing environment")
+	}
+	if !common.IsHexAddress(*c.CoordinatorAddress) {
+		return errors.New("coordinator_address must be a valid hex address")
+	}
+	if c.KeyHash == nil {
+		return errors.New("key_hash must be set when using existing environment")
+	}
+	if *c.KeyHash == "" {
+		return errors.New("key_hash must be a non-empty string")
+	}
+	if c.LinkAddress != nil && !common.IsHexAddress(*c.LinkAddress) {
+		return errors.New("link_address must be a valid hex address")
+	}
+
+	if *c.CreateFundSubsAndAddConsumers {
+		if err := c.Funding.Validate(); err != nil {
+			return err
+		}
+	} else {
+		if c.ConsumerAddress == nil || *c.ConsumerAddress == "" {
+			return errors.New("consumer_address must be set when using existing environment")
+		}
+		if !common.IsHexAddress(*c.ConsumerAddress) {
+			return errors.New("consumer_address must be a valid hex address")
+		}
+	}
+
+	if c.NodeSendingKeys != nil {
+		for _, key := range c.NodeSendingKeys {
+			if !common.IsHexAddress(key) {
+				return errors.New("node_sending_keys must be a valid hex address")
+			}
+		}
+	}
+
+	return nil
+}
+
+type NewEnvConfig struct {
```

Comment:
we don't use it anymore, should we remove it?

Reply:
yes, will do